stock_auto_move/models/stock_move.py

```diff
@@ -30,10 +30,23 @@ def _get_auto_moves_by_pickings(self, auto_moves):
 
     @api.multi
     def action_assign(self, no_prepare=False):
-        res = super(StockMove, self).action_assign(no_prepare=no_prepare)
+
+        already_assigned_moves = self.filtered(
+            lambda m: m.state == 'assigned')
+
+        # If all the moves have been already been assigned return the
+        # super method
+        if len(already_assigned_moves) == len(self):
```

Comment:
This is not necessary as the result recordset here below (not_assigned_auto_move) will be void.
Maybe you don't want to enter the below functions but that's to be put in balance with the amount of code

Reply:
@rousseldenis thanks for the review, and it's fixed :)